qutebrowser/mainwindow/tabwidget.py

```diff
@@ -457,12 +450,35 @@ def minimumTabSizeHint(self, index):
             padding_h += self.style().pixelMetric(
                 PixelMetrics.icon_padding, None, self)
         height = self.fontMetrics().height() + padding_v
-        width = (self.fontMetrics().width('\u2026') + icon_size.width() +
+        width = (self.fontMetrics().width(text) + icon_size.width() +
                  padding_h + config.val.tabs.width.indicator)
         return QSize(width, height)
 
-    def tabSizeHint(self, index):
-        """Override tabSizeHint so all tabs are the same size.
+    def _tabTotalWidthPinned(self):
+        """Get the current total width of pinned tabs.
+
+        This width is calculated assuming no shortening due to elipsis."""
+        return sum(
+            # Get the width (int) from minimum size
+            map(lambda tab: tab.width(),
+                # Get the minimum size of tabs, no elipsis
+                map(functools.partial(self.minimumTabSizeHint, elipsis=False),
+                    # Get only pinned tabs (from indexes)
+                    filter(self._tabPinned, range(self.count())))))
```

Comment:
I think this functional style of expressing this is rather hard to read - I'd prefer (untested):

```python
return sum(self.minimumTabSizeHint(idx, ellipsis=False).width()
           for idx in range(self.count())
           if self._tab_pinned(idx))
```

Reply:
I actually prefer the opposite, I think that list comprehension obfuscates the flow of data.

For example, the data in the example above starts at line 2, is filtered by line 3, and is transformed by line 1 (which forces people understanding the code to jump around between the lines to understand how the data is manipulated), whereas in the functional example the data flows from the last line to the first. 

Unfortunately, the data flow is backwards, because python doesn't have something similar to a threading macro.

For consistency though, I'll change it to use list comprehensions instead :cry: .